### ISSUE TITLE
Incorrect highlighting of repetitions in macro_rules

### ISSUE THREAD

**Issue by agausmann** [rust-lang/rust.vim#405]
* rust.vim version: 
  - Latest `master` at the time of writing (2b916c8c), installed manually at `$HOME/.vim/pack/git-plugins/start/rust.vim`.

* Steps to reproduce:
  - Create a file `foo.rs` file with the contents:

  ```rust
  macro_rules! foo {
      ($(($x:ident)),*) => {
          $(struct $x();)*
      }
  }
  ```

* Expected vs. actual behavior:
  - Repetitions in `macro_rules` macros aren't highlighted properly. The first closing parentheses found is treated as the end of the repetition, which is not always correct.

  Example - Here is the incorrect highlighting created by `rust.vim`:
  
  ![2020-05-13-140629_390x230_scrot](https://user-images.githubusercontent.com/6611767/81854194-2f689b00-9523-11ea-915e-15904f1ab36f.png)

  Here is the expected highlighting, which is what Vim does without any configuration or plugins:

  ![2020-05-13-140947_390x230_scrot](https://user-images.githubusercontent.com/6611767/81854431-88d0ca00-9523-11ea-8737-a70b9d9e4e8c.png)

* `RustInfo`
```
rust.vim Global Variables:

let g:ftplugin_rust_source_path = v:null
let g:loaded_syntastic_rust_cargo_checker = v:null
let g:loaded_syntastic_rust_filetype = v:null
let g:loaded_syntastic_rust_rustc_checker = v:null
let g:rust_bang_comment_leader = v:null
let g:rust_cargo_avoid_whole_workspace = v:null
let g:rust_clip_command = v:null
let g:rust_conceal = v:null
let g:rust_conceal_mod_path = v:null
let g:rust_conceal_pub = v:null
let g:rust_fold = v:null
let g:rust_last_args = v:null
let b:rust_last_args = []
let g:rust_last_rustc_args = v:null
let b:rust_last_rustc_args = []
let g:rust_original_delimitMate_excluded_regions = v:null
let g:rust_playpen_url = v:null
let g:rust_prev_delimitMate_quotes = v:null
let g:rust_recent_nearest_cargo_tol = v:null
let g:rust_recent_root_cargo_toml = v:null
let g:rust_recommended_style = v:null
let g:rust_set_conceallevel = v:null
let g:rust_set_conceallevel=1 = v:null
let g:rust_set_foldmethod = v:null
let g:rust_set_foldmethod=1 = v:null
let g:rust_shortener_url = v:null
let g:rustc_makeprg_no_percent = v:null
let g:rustc_path = v:null
let g:rustfmt_autosave = 1
let g:rustfmt_autosave_if_config_present = v:null
let g:rustfmt_command = 'rustfmt'
let g:rustfmt_emit_files = 1
let g:rustfmt_fail_silently = 0
let g:rustfmt_options = ''
let g:syntastic_extra_filetypes = ['rust']
let g:syntastic_rust_cargo_fname = v:null
rustfmt 1.4.14-nightly (a5cb5d2 2020-04-14)

rustc 1.45.0-nightly (769d12eec 2020-05-12)

cargo 1.45.0-nightly (cb06cb269 2020-05-08)


VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Apr 28 2020 13:09:29)
Included patches: 1-654
Compiled by void-buildslave@a-hel-fi
Huge version without GUI.  Features included (+) or not (-):
+acl               +cursorshape       +job               +mouse_urxvt       +signs             +vartabs
+arabic            +dialog_con        +jumplist          +mouse_xterm       +smartindent       +vertsplit
+autocmd           +diff              +keymap            +multi_byte        -sound             +virtualedit
+autochdir         +digraphs          +lambda            +multi_lang        +spell             +visual
-autoservername    -dnd               +langmap           -mzscheme          +startuptime       +visualextra
-balloon_eval      -ebcdic            +libcall           -netbeans_intg     +statusline        +viminfo
+balloon_eval_term +emacs_tags        +linebreak         +num64             -sun_workshop      +vreplace
-browse            +eval              +lispindent        +packages          +syntax            +wildignore
++builtin_terms    +ex_extra          +listcmds          +path_extra        +tag_binary        +wildmenu
+byte_offset       +extra_search      +localmap          -perl              -tag_old_static    +windows
+channel           -farsi             -lua               +persistent_undo   -tag_any_white     +writebackup
+cindent           +file_in_path      +menu              +popupwin          -tcl               -X11
-clientserver      +find_in_path      +mksession         +postscript        +termguicolors     -xfontset
-clipboard         +float             +modify_fname      +printer           +terminal          -xim
+cmdline_compl     +folding           +mouse             +profile           +terminfo          -xpm
+cmdline_hist      -footer            -mouseshape        -python            +termresponse      -xsmp
+cmdline_info      +fork()            +mouse_dec         -python3           +textobjects       -xterm_clipboard
+comments          +gettext           -mouse_gpm         +quickfix          +textprop          -xterm_save
+conceal           -hangul_input      -mouse_jsbterm     +reltime           +timers
+cryptv            +iconv             +mouse_netterm     +rightleft         +title
+cscope            +insert_expand     +mouse_sgr         -ruby              -toolbar
+cursorbind        +ipv6              -mouse_sysmouse    +scrollbind        +user_commands
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
  fall-back for $VIM: "/usr/share/vim"
Compilation: cc -c -I. -Iproto -DHAVE_CONFIG_H     -fstack-clash-protection -mtune=generic -O2 -pipe   -g -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1
Linking: cc   -Wl,-z,relro -Wl,-z,now     -L/usr/local/lib -Wl,--as-needed -o vim        -lm -lelf  -lncursesw -lacl -lattr -ldl
```

**Comment by agausmann** [comment]
Closed by #396

**Comment by da-x** [comment]
Merged.
